packages/cli/src/dbt/context.ts

```diff
@@ -0,0 +1,40 @@
+import { ParseError } from 'common';
+import { readFileSync } from 'fs';
+import * as yaml from 'js-yaml';
+import * as path from 'path';
+
+type GetDbtContextArgs = {
+    projectDir: string;
+};
+type DbtContext = {
+    projectName: string;
+    profileName: string;
+    targetDir: string;
+    modelsDir: string;
+};
+
+export const getDbtContext = ({
+    projectDir,
+}: GetDbtContextArgs): DbtContext => {
+    const projectFilename = path.join(projectDir, 'dbt_project.yml');
+    let config;
+    try {
+        config = yaml.load(
+            readFileSync(projectFilename, { encoding: 'utf-8' }),
+        ) as any;
+    } catch (e) {
+        throw new ParseError(
+            `Is ${projectDir} a valid dbt project? Couldn't find a valid dbt_project.yml file:\n${e.message}`,
+        );
+    }
+    const targetSubDir = config['target-path'] || './target';
```

Comment:
Should we support target-dir and models-dir ? We support it in `dbtCliClient.ts` 

`config['target-path'] || config['target-dir'] || '/target',`

Reply:
This was recently renamed! I'll update this

Reply:
https://docs.getdbt.com/reference/dbt_project.yml